sktime/forecasting/hf_momentfm_forecaster.py

```diff
@@ -0,0 +1,780 @@
+        If the sequence length is > 512, it will be reduced to 512.
+        default = 512
+
+    batch_size : int
+        size of batches to train the model on
+        default = 32
+
+    eval_batch_size : int or "all"
+        size of batches to evaluate the model on. If the string "all" is
+        specified, then we process the entire validation set as a single batch
+        default = 32
+
+    epochs : int
+        Number of epochs to fit tune the model on
+        default = 1
+
+    max_lr : float
+        Maximum learning rate that the learning rate scheduler will use
+        default = 1e-4
+
+    device : str
+        torch device to use
+        default = gpu
+
+    pct_start : float
+        percentage of total iterations where the learning rate rises during
+        one epoch
+        default = 0.3
+
+    max_norm : float
+        Float value used to clip gradients during training
+        default = 5.0
+
+    train_val_split : float
+        float value between 0 and 1 to determine portions of training
+        and validation splits
+        default = 0.4
+
+    transformer_backbone : str
+        d_model of a pre-trained transformer model to use. See
+        SUPPORTED_HUGGINGFACE_MODELS to specify valid models to use.
+        Default is 'google/flan-t5-large'.
+
+    config : dict, default = {}
+        If desired, user can pass in a config detailing all momentfm parameters
+        that they wish to set in dictionary form, so that parameters do not need
+        to be individually set. If a parameter inside a config is a
+        duplicate of one already passed in individually, it will be overwritten.
+
+    return_model_to_cpu : bool, default = False
+        After fitting and training, will return the `momentfm` model to the cpu.
+    """
+
+    _tags = {
+        "scitype:y": "both",
+        "authors": ["julian-fong"],
+        "maintainers": ["julian-fong"],
+        "handles-missing-data": False,
+        "y_inner_mtype": [
+            "pd.DataFrame",
+            "pd-multiindex",
+            "pd_multiindex_hier",
+        ],
+        "ignores-exogeneous-X": True,
+        "requires-fh-in-fit": True,
+        "python_dependencies": [
+            "torch",
+            "tqdm",
+            "huggingface-hub",
+            # "momentfm",
+            "accelerate",
+        ],
+        "capability:global_forecasting": True,
+        "python_version": ">= 3.10",
+        "capability:insample": False,
+        "capability:pred_int:insample": False,
+        "capability:pred_int": False,
+    }
+
+    def __init__(
+        self,
+        pretrained_model_name_or_path="AutonLab/MOMENT-1-large",
+        freeze_encoder=True,
+        freeze_embedder=True,
+        freeze_head=False,
+        dropout=0.1,
+        head_dropout=0.1,
+        seq_len=512,
+        batch_size=32,
+        eval_batch_size=32,
+        epochs=1,
+        max_lr=1e-4,
+        device="gpu",
```

Comment:
is there an option "auto"? Which detects automatically which kind of accelerator/device is available? 

If yes I would suggest to use it so that the default config is also working for MacOS etc.